### PR TITLE
Adds initial capability for managing a yum repo of rebuilt packages from amzn linux

### PR DESCRIPTION
**Pull request by lorengordon**
This supports retrieving srpm packages from Amzn Linux 2, rebuilding the packages for EL7 and EL8, generating a yum repo of those packages, and syncing the yum repo to an S3 bucket.

Uses multi-stage docker containers to retrieve srpms and build packages for EL7 and EL8. The basic workflow is something like this:

 1. Build the docker image using Dockerfile.elN
 2. Run the container from that image
 3. Copy the built packages to the host
 4. Sign the packages
 5. Copy the signed packages from the host to the running container
 6. Create the repodata in the container
 7. Copy the repodata back to the host
 8. Sign the repodata
 9. Copy the signed repodata to the running container
 10. Test the repo
 11. Push the repo to S3 bucket

The github actions file `build.yml` implements that workflow. The GPG key is stored as a GitHub Secret. For pull requests from forks, secrets are not available, so the signing steps are skipped and the repo is tested with unsigned packages/repodata.

### CODE REVIEW

**Comment by lorengordon** [comment]
@plus3it/spel This is ready for a first look...

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+jobs:
+  BuildRepo:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        el_version: [7, 8]
+    env:
+      AWS_DEFAULT_REGION: us-east-1
+      REPO_ENDPOINT: https://spel-packages.cloudarmor.io
```
Comment (ferricoxide) [inline .github/workflows/build.yml:29]: Similar to region-related comments and future flexibility

File: Dockerfile.el7 [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        ant \
+        dos2unix \
+        java-1.8.0-openjdk \
+        make \
+        python2-devel \
+        python3 \
+        pytest
+
+# Install requirements for building amznlinux srpms from epel repos
+RUN yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm \
```
Comment (ferricoxide) [inline Dockerfile.el7:43]: See other hardcode comments

File: Dockerfile.el8 [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        ant \
+        dos2unix \
+        java-1.8.0-openjdk \
+        make \
+        python2-devel \
+        python3 \
+        python3-pytest
+
+# Install requirements for building amznlinux srpms from epel repos
+RUN yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm \
```
Comment (ferricoxide) [inline Dockerfile.el8:43]: Same as for the el7 file

**Review by ferricoxide — COMMENTED** [review]
Overall, my question would be around rehosting flexibility (seemed redundant to add questions for the hardcoded container-IDs): "is this only ever to be run from/for CONUS regions? Are the resultant packages going to contain anything that's apt to make them non-portable (if it's intended that packages created in this project will be flowed to other partitions/locations)?"

File: Dockerfile.el7 (hunk 2 above)
Comment (lorengordon) [reply Dockerfile.el7:43]: Yeah, these can be a build ARG

**Comment by lorengordon** [comment]
@ferricoxide The github-actions are intended to be localized, definitely not portable. The Dockerfiles ought to remain reasonably portable though.

But, the first approach for portability is to mirror the yum repo, excluding the spel-release package, and use a custom release rpm to point to wherever it is re-hosted. If that doesn't work out, we'll revisit making more of the overall workflow more portable.

**Review by eemperor — COMMENTED** [review]
You have a couple `.gitkeep` stragglers in the PR:

`.repo/el7/.gitkeep`
`.repo/el8/.gitkeep`

**Comment by lorengordon** [comment]
> You have a couple `.gitkeep` stragglers in the PR:
> 
> `.repo/el7/.gitkeep` `.repo/el8/.gitkeep`

@eemperor Ok, I removed those. They were necessary previously, to ensure the `.repo/el{7,8}/` paths existed, but I went ahead and precreated them in the github action. Thanks, this makes for a cleaner git repo!

**Review by ferricoxide — APPROVED** [review]
In-project and out-of-band conversations resolved.

**Pull request merged**